pkg/assetattributor/cloudassetinventory.go

```diff
@@ -85,16 +85,12 @@ type CloudAssetInventory struct {
 // Attribute queries the asecurityteam/asset-inventory-api service first by IP, then by hostname
 // if the first query returns no results
 func (n *CloudAssetInventory) Attribute(ctx context.Context, asset domain.NexposeAssetVulnerabilities) (domain.NexposeAttributedAssetVulnerabilities, error) {
-	if asset.ScanTime.IsZero() {
```

Comment:
Question: why remove this check?

Reply:
because asset-producer guarantees this field to be nonzero

Reply:
You should keep the check though.  This service (asset-attributor) should not assume all communication originates from asset-producer, nor should its defense of the payload contract be reduced.

Reply:
i'll put it back

Reply:
There a variety of factors in play here. I fully agree with @mikerott that no code should ever be written with assumptions about what happens on the other side of a service boundary, both to ensure program behavioral correctness, and for maintainability of the code.

However! The OpenAPI spec for this attribute is required to be a non-empty string in RFC 3339 datetime format, and so this much at least is actually enforced at this service's boundary, and so it would seem that the check here is redundant.

And yet! The thrilling drama concludes with this: in go, `time.IsZero` does not check for a value of empty string, but for the zero value of a `time` instance, which is `0001-01-01 00:00:00+0000 UTC`. This means that the zero value - which is absolutely an invalid value that should be rejected - could be validated by the transportd gateway and reach this code path, making the check necessary.

TL;DR: the check **should** remain, but it's important that everyone understand why.